library/core/src/ptr/non_null.rs

```diff
@@ -762,6 +768,7 @@ impl<T: ?Sized> NonNull<T> {
     #[cfg_attr(miri, track_caller)] // even without panics, this helps for Miri backtraces
     #[stable(feature = "non_null_convenience", since = "1.80.0")]
     #[rustc_const_stable(feature = "non_null_convenience", since = "1.80.0")]
+    #[ensures(|result: &isize| *result == (self.as_ptr() as isize - origin.as_ptr() as isize) / core::mem::size_of::<T>() as isize)]
     pub const unsafe fn offset_from(self, origin: NonNull<T>) -> isize
```

Comment:
Can you add some safety preconditions for this function? https://doc.rust-lang.org/beta/std/ptr/struct.NonNull.html#method.offset_from

Reply:
Also, these docs say that the function panics if `T` is a ZST, so can you add a [`#[should_panic]` harness ](https://model-checking.github.io/kani/reference/attributes.html?highlight=panic#kanishould_panic)checking that?

Reply:
all done